docs/OV_Converter_UG/prepare_model/convert_model/Convert_Model_From_TensorFlow.md

```diff
@@ -0,0 +1,311 @@
+
+         ovc path_to_inference_graph.pb path_to_checkpoint_file.ckpt
+
+To convert the model with the inference graph in ``.pbtxt`` format, specify the path to ``.pbtxt`` file instead of ``.pb`` file. The conversion API automatically detects the format of the provided file, there is no need to specify the model file format explicitly when calling `ovc` or `openvino.convert_model` in all examples in this document.
+
+3. **MetaGraph**. In this case, a model consists of three or four files stored in the same directory: ``model_name.meta``, ``model_name.index``,
+``model_name.data-00000-of-00001`` (the numbers may vary), and ``checkpoint`` (optional).
+To convert such a TensorFlow model, run the conversion providing a path to `.meta` file as an argument:
+
+.. tab-set::
+
+   .. tab-item:: Python
+      :sync: py
+
+         import openvino as ov
+         ov_model = ov.convert_model('path_to_meta_graph.meta')
+
+   .. tab-item:: CLI
+      :sync: cli
+
+         ovc path_to_meta_graph.meta
+
+
+Freezing Custom Models in Python
+++++++++++++++++++++++++++++++++
+
+When a network is defined in Python code, you must create an inference graph file. Graphs are usually built in a form
+that allows model training. That means all trainable parameters are represented as variables in the graph.
+To be able to use such a graph with model conversion API, it should be frozen first before passing to ``openvino.convert_model`` function:
+
+** TODO: Check correctness, provide complete examples (here and below) **
+
+.. code-block:: py
+   :force:
+
+   import tensorflow as tf
+   from tensorflow.python.framework import graph_io
+   frozen = tf.compat.v1.graph_util.convert_variables_to_constants(sess, sess.graph_def, ["name_of_the_output_node"])
+
+   import openvino as ov
+   ov_model = ov.convert_model(frozen)
+
+Where:
+
+* ``sess`` is the instance of the TensorFlow Session object where the network topology is defined.
+* ``["name_of_the_output_node"]`` is the list of output node names in the graph; ``frozen`` graph will include only those nodes from the original ``sess.graph_def`` that are directly or indirectly used to compute given output nodes. The ``'name_of_the_output_node'`` is an example of a possible output node name. You should derive the names based on your own graph.
+
+Conversion of TensorFlow models from memory using Python API
+############################################################
+
+Model conversion API supports passing TensorFlow/TensorFlow2 models directly from memory.
+
+* ``tf.keras.Model``
+
+  .. code-block:: py
+     :force:
+
+     import openvino as ov
+     model = tf.keras.applications.ResNet50(weights="imagenet")
+     ov_model = ov.convert_model(model)
+
+* ``tf.keras.layers.Layer``. Requires setting the "input_shape".
+
+  .. code-block:: py
+     :force:
+
+     import tensorflow_hub as hub
+     import openvino as ov
+
+     model = hub.KerasLayer("https://tfhub.dev/google/imagenet/mobilenet_v1_100_224/classification/5")
```

Comment:
in example above, we require to save model on disk before conversion since we have a problem with input/output names and accuracy